racket/src/racket/src/network.c

```diff
@@ -1457,6 +1459,81 @@ void scheme_tcp_abandon_port(Scheme_Object *port) {
   tcp_abandon_port(1, &port);
 }
 
+static Scheme_Object *tcp_sendfile(int argc, Scheme_Object *argv[]) {
+  Scheme_Tcp *data;
+  Scheme_Input_Port *ip;
+  Scheme_Output_Port *op;
+  rktio_fd_t *src_fd;
+  intptr_t sent;
+  intptr_t offset = -1;
+  intptr_t nbytes = -1;
+  if (SCHEME_OUTPUT_PORTP(argv[0])) {
+    op = scheme_output_port_record(argv[0]);
+    if (op->sub_type == scheme_tcp_output_port_type) {
+      if (op->closed) {
+        scheme_raise_exn(MZEXN_FAIL_NETWORK, "tcp-sendfile: tcp port closed");
+        return NULL;
+      }
+
+      if (SCHEME_INPUT_PORTP(argv[1])) {
+        ip = scheme_input_port_record(argv[1]);
+        if (SCHEME_FALSEP(scheme_file_stream_port_p(1, &ip))) {
+          scheme_wrong_contract("tcp-sendfile", "(and/c file-stream-port? input-port?)", 1, argc, argv);
+          return NULL;
+        }
+
+        if (argc > 2) {
+          if (SCHEME_INTP(argv[2])) {
```

Comment:
`SCHEME_INTP` recognizes only fixnums, but `exact-nonnegative-integer?` includes bignums. Figuring out what to do with bignums is always a pain, but that's something to work out here, too.